allennlp/predictors/event2mind.py

```diff
@@ -0,0 +1,22 @@
+from overrides import overrides
+
+from allennlp.common.util import JsonDict
+from allennlp.data import Instance
+from allennlp.predictors.predictor import Predictor
+
+@Predictor.register('event2mind')
+class Event2MindPredictor(Predictor):
+    """
+    Predictor for the :class:`~allennlp.models.event2mind` model.
+    """
+
+    def predict(self, source: str) -> JsonDict:
```

Comment:
Good call, done.

It occurs to me that there is some tight coupling between these predictors and the model itself. Perhaps we should make this coupling explicit by adding the predictors to the API for the model classes.

Reply:
Super-helpful comment.  Potentially related to your point: https://github.com/allenai/allennlp/issues/1675

The motivation was that we might have multiple predictors with different formats per model.  Personally, I'd welcome a default "predict" method on the model itself.

Reply:
Yeah, the motivation makes sense. It's just a matter of having a clean API for that motivation... I'll ponder some more later.

Reply:
I think it would only be json to json. That's another shift that's happened over time. Predictors used to only be for file output but now they can also be used programmatically.

Reply:
I think it would only be json to json. That's another shift that's happened over time. Predictors used to only be for file output but now they can also be used programmatically.

Reply:
@matt-gardner, my point is really just that this predictor as well as the `simple_seq2seq` one are effectively just shims. They don't do anything other than dispatch to the `DatasetReader` and the `Model`. While this does not appear to be true generally, it suggests that we could provide a default predictor automatically in the simple case. No need for a `model.predict()`. This is indeed related to your https://github.com/allenai/allennlp/issues/1675 as @schmmd pointed out.

Reply:
Yeah, I agree that it'd be nice to do something smarter here for easy cases.  I was just clarifying some API decisions about why it's set up the way that it is.

And @schmmd, json to json still doesn't solve the problem.  The way the `Predictor` handles json to json is by delegating to the `DatasetReader` - it's the `DatasetReader` that actually knows how to create the tensors that get input to the model.  The `Model` can't do json to json by itself without access to the `DatasetReader`.

Reply:
And the clarification is appreciated! :)